### PR TITLE
Translations update from Hosted Weblate

### PR DESCRIPTION
**Pull request by weblate**
Translations update from [Hosted Weblate](https://hosted.weblate.org) for [Hedy/Adventures](https://hosted.weblate.org/projects/hedy/adventures/).

It also includes following components:

* [Hedy/client-messages](https://hosted.weblate.org/projects/hedy/client-messages/)

* [Hedy/Webpages](https://hosted.weblate.org/projects/hedy/webpages/)

* [Hedy/Parsons](https://hosted.weblate.org/projects/hedy/parsons/)

* [Hedy/Quizzes](https://hosted.weblate.org/projects/hedy/quizzes/)

* [Hedy/Texts](https://hosted.weblate.org/projects/hedy/web-texts/)

* [Hedy/Achievements](https://hosted.weblate.org/projects/hedy/achievements/)

* [Hedy/Commands](https://hosted.weblate.org/projects/hedy/commands/)

* [Hedy/Tutorials](https://hosted.weblate.org/projects/hedy/tutorials/)

* [Hedy/Keywords](https://hosted.weblate.org/projects/hedy/keywords/)

Current translation status:

Weblate translation status

### CODE REVIEW

File: content/adventures/tn.yaml [hunk 1]
```diff
@@ -63,9 +63,9 @@ adventures:
...
             10:
-                start_code: "animals {is} dog, cat, blobfish\n{for} animal {in} animals\n  {print} 'I love ' animal"
-                story_text: "## For each item in my list...\nIn this level we learn a new code called `{for}`. With `{for}` you can make a list and use all elements.\n`{for}` creates a block, like `{repeat}` and `{if}` so all lines in the block need to start with spaces."
-                example_code: "```\nanimals {is} dog, cat, blobfish\n{for} animal {in} animals\n  {print} 'I love ' animal\n```\n"
+                start_code: "Diphologolo {is} ntsa, katse, thapi\n{for} phologolo {in} diphologolo\n  {print} 'ke rata ' phologolo"
```
Comment (Felienne) [inline content/adventures/tn.yaml:66]:
```suggestion
                start_code: "diphologolo {is} ntsa, katse, thapi\n{for} phologolo {in} diphologolo\n  {print} 'ke rata ' phologolo"
```

File: content/commands/uk.yaml [hunk 2]
```diff
@@ -31,171 +31,171 @@
...
-    demo_code: "{print} '5 times 5 is ' 5 * 5"
+    explanation: Видрукуйте щось. Памʼятайте вживати лапки для друку конкретних фраз.
+    demo_code: "{print} '5 на 5 це ' 5 * 5"
 -   name: '{ask}'
-    explanation: Ask for the answer to a sum and check if it is correct. We can now print 2 lines.
-    demo_code: "answer = {ask} 'What is 5 plus 5?'\n{if} answer {is} 10\n    {print} 'Well done!'\n    {print} 'Indeed, the answer was ' answer\n{else}\n    {print} 'Oops!'\n    {print} 'The answer is 10'"
--   name: '{repeat} with turtle'
-    explanation: Repeat multiple lines.
+    explanation: Запитайте про суму і перевірте відповідь. Тепер ми можемо друкувати дві лінії.
+    demo_code: "відповідь = {ask} 'Скільки буде 5 додати 5?'\n{if} відповідь {is} 10\n    {print} 'Добра робота!'\n    {print} 'Дійсно, правильна відповідь ' answer\n{else}\n    {print} 'Ой!'\n    {print} 'Відповідь 10'"
```
Comment (Felienne) [inline content/commands/uk.yaml:98]:
```suggestion
    demo_code: "відповідь = {ask} 'Скільки буде 5 додати 5?'\n{if} відповідь {is} 10\n    {print} 'Добра робота!'\n    {print} 'Дійсно, правильна відповідь ' відповідь\n{else}\n    {print} 'Ой!'\n    {print} 'Відповідь 10'"
```

File: content/commands/uk.yaml [hunk 3]
```diff
@@ -31,171 +31,171 @@
...
+    explanation: Повторіть кілька рядків.
     demo_code: "{repeat} 4 {times}\n    {turn} 90\n    {forward} 50"
 10:
 -   name: '{print}'
-    explanation: Print something. Remember to use a quotation mark for literal printing.
-    demo_code: "{print} '5 times 5 is ' 5 * 5"
--   name: '{for} with a list'
-    explanation: Print all things in a list.
+    explanation: Надрукуйте щось. Не забувайте використовувати лапки для літерного друку.
+    demo_code: "{print} '5 помножити на 5' - це 5 * 5"
```
Comment (Felienne) [inline content/commands/uk.yaml:112]:
```suggestion
    demo_code: "{print} '5 помножити на 5 - це' 5 * 5"
```

File: content/keywords/uk.yaml [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
 else: інакше
 and: і
 0: '0'
-repeat: повторити
-times: раз
+repeat: повтори
+times: разів|рази|раз
 for: для
-range: діапазон
+range: діапазоні|діапазон
```
Comment (Felienne) [inline content/keywords/uk.yaml:14]:
```suggestion
range: діапазон|діапазонi
```

**Comment by Felienne** [comment]
Wow, it seems Punjabi is not a language that Babel recognizes?! That seems very weird with over 100 million speakers (I mean it was weird for Tswana but this is next level...) 

I will a) skip this language for now and b) look deeper into the issue maybe the Pakistani is the problem.

@TiBiBa Did we ever hear back from babel about Tswana?

**Comment by TiBiBa** [comment]
> Wow, it seems Punjabi is not a language that Babel recognizes?! That seems very weird with over 100 million speakers (I mean it was weird for Tswana (see #3141) but this is next level...)
> 
> I will a) skip this language for now and b) look deeper into the issue maybe the Pakistani is the problem.
> 
> @TiBiBa Did we ever hear back from babel about Tswana?

I did! However, they decided to close the issue here python-babel/babel#899

> Unfortunately this needs to be fixed via the CLDR.
>
>Adding support for project-local arbitrary languages has been talked about in the past (https://github.com/python->babel/babel/issues/454) but it requires quite some refactoring.

**Comment by Felienne** [comment]
> > Wow, it seems Punjabi is not a language that Babel recognizes?! That seems very weird with over 100 million speakers (I mean it was weird for Tswana (see #3141) but this is next level...)
> > I will a) skip this language for now and b) look deeper into the issue maybe the Pakistani is the problem.
> > @TiBiBa Did we ever hear back from babel about Tswana?
> 
> I did! However, they decided to close the issue here [python-babel/babel#899](https://github.com/python-babel/babel/issues/899)
> 
> > Unfortunately this needs to be fixed via the CLDR.
> > Adding support for project-local arbitrary languages has been talked about in the past ([https://github.com/python->babel/babel/issues/454](https://github.com/python-%3Ebabel/babel/issues/454)) but it requires quite some refactoring.

Ah I see they rely on a standard which apparently does not recognise Tswana (bit much to call this a "project-local arbitrary language" but ok, I understand their approach) while Weblate doen recognise it. We should maybe petition at CLDR then, I will try!

As for Punjabi there is a solution (not a pretty one but ok)

**Comment by Felienne** [comment]
> > > Wow, it seems Punjabi is not a language that Babel recognizes?! That seems very weird with over 100 million speakers (I mean it was weird for Tswana (see #3141) but this is next level...)
> > > I will a) skip this language for now and b) look deeper into the issue maybe the Pakistani is the problem.
> > > @TiBiBa Did we ever hear back from babel about Tswana?
> > 
> > 
> > I did! However, they decided to close the issue here [python-babel/babel#899](https://github.com/python-babel/babel/issues/899)
> > > Unfortunately this needs to be fixed via the CLDR.
> > > Adding support for project-local arbitrary languages has been talked about in the past ([https://github.com/python->babel/babel/issues/454](https://github.com/python-%3Ebabel/babel/issues/454)) but it requires quite some refactoring.
> 
> Ah I see they rely on a standard which apparently does not recognise Tswana (bit much to call this a "project-local arbitrary language" but ok, I understand their approach) while Weblate doen recognise it. We should maybe petition at CLDR then, I will try!

Hmmmmm it is weird though, CLDR does seem to list Tswana:

https://github.com/unicode-org/cldr/blob/62c90a357dc25911db60fcdf7d5a80119df27963/tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/language_list.txt

I will reply on the Babel issue

**Comment by Felienne** [comment]
Also, just noting it down so we don't forget it, their original issue has a (complex) patch idea that we could also try: https://github.com/python-babel/babel/issues/454 (although I prefer petitioning with/solving in Babel so others can benefit too)

**Comment by Felienne** [comment]
Thanks for a new language @bgo-eiu! Should be live on alpha for you to look at in a few minutes!

**Review by Felienne — APPROVED** [review]

**Pull request merged**